### PR TITLE
http://elastic.co -> https

### PR DESCRIPTION
**Pull request by jbudz**
Quick find and replace from `http` to https` for references to elastic.co. Mostly localized to the rest spec.

### CODE REVIEW

**Comment by dnhatn** [comment]
@jbudz Thanks for picking this up. There is a test failure that relates to your changes. Can you please fix it?

**Comment by ywelsch** [comment]
@elastic/docs can someone from your team take a look here?

**Comment by jrodewig** [comment]
@ywelsch  I can review this. Thanks!

File: docs/plugins/discovery-ec2.asciidoc [hunk 1]
```diff
@@ -8,7 +8,7 @@ matching certain criteria determined by the <<discovery-ec2-usage,plugin
 settings>>.
 
 *If you are looking for a hosted solution of {es} on AWS, please visit
-http://www.elastic.co/cloud.*
+https://www.elastic.co/cloud.*
```
Comment (jrodewig) [inline docs/plugins/discovery-ec2.asciidoc]: It looks like we enforce a trailing slash here.
```suggestion
https://www.elastic.co/cloud/.*
```

File: docs/plugins/redirects.asciidoc [hunk 2]
```diff
@@ -13,7 +13,7 @@ one of the <<discovery,cloud discovery plugins>>.
 [role="exclude",id="cloud-aws"]
 === AWS Cloud Plugin
 
-Looking for a hosted solution for Elasticsearch on AWS? Check out http://www.elastic.co/cloud.
+Looking for a hosted solution for Elasticsearch on AWS? Check out https://www.elastic.co/cloud.
```
Comment (jrodewig) [inline docs/plugins/redirects.asciidoc]:
```suggestion
Looking for a hosted solution for Elasticsearch on AWS? Check out https://www.elastic.co/cloud/.
```/\

File: docs/plugins/repository-s3.asciidoc [hunk 3]
```diff
@@ -5,7 +5,7 @@ The S3 repository plugin adds support for using AWS S3 as a repository for
 {ref}/modules-snapshots.html[Snapshot/Restore].
 
 *If you are looking for a hosted solution of Elasticsearch on AWS, please visit
-http://www.elastic.co/cloud.*
+https://www.elastic.co/cloud.*
```
Comment (jrodewig) [inline docs/plugins/repository-s3.asciidoc]:
```suggestion
https://www.elastic.co/cloud/.*
```

File: x-pack/plugin/src/test/resources/rest-api-spec/api/ccr.forget_follower.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "ccr.forget_follower":{
     "documentation":{
-      "url":"http://www.elastic.co/guide/en/elasticsearch/reference/current"
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current"
```
Comment (jrodewig) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/ccr.forget_follower.json]: Not related to your changes, but this link is wrong.
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-forget-follower.html"
```

File: x-pack/plugin/src/test/resources/rest-api-spec/api/ccr.unfollow.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "ccr.unfollow":{
     "documentation":{
-      "url":"http://www.elastic.co/guide/en/elasticsearch/reference/current"
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current"
```
Comment (jrodewig) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/ccr.unfollow.json]: Link is wrong.
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-unfollow.html"
```

**Comment by jrodewig** [comment]
@jbudz  Thanks for this PR. I added a few suggestions, most of which weren't directly related to your changes.

It looks like you also have some merge conflicts on this PR.

Once those two issues are addressed, I can help merge and backport these changes. Let me if you need any help implementing the feedback.

**Review by jrodewig — CHANGES_REQUESTED** [review]
@jbudz Thanks for this PR. I added a few suggestions, most of which weren't directly related to your changes.

It looks like you also have some merge conflicts on this PR.

Once those two items are addressed, I can help merge and backport these changes. Let me if you need any help implementing the feedback.

**Review by jrodewig — APPROVED** [review]
Thanks @jbudz

**Comment by jrodewig** [comment]
### Backport commits
master 23b31d6
7.x 4ea7297

**Pull request merged**